---
title: "FigureS6_plots"
author: "Miranda Hunter"
date: "5/14/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Plots from Figure S6 of Hunter, Moncada et al.

Libraries:
``` {r, message = F, warning = F}
library(Seurat)
library(tidyverse)
library(scales)
library(pals)
library(cowplot)
library(fgsea)
library(msigdbr)
library(mvhspatialplots)
```

Load snRNA-seq dataset:
``` {r}
load('/Users/hunterm/Dropbox/10x/nuc_seq/NS_norm_new_hashing_nodoublets.R')
load("/Users/hunterm/Dropbox/10x/R_objects/EF_integrated_allgenes_new.R")
```


Figure S6a: UMIs and genes per nucleus
``` {r}
NS_new_metadata <- NS_new[[]]

p2 <- ggplot(NS_new_metadata, aes(x = nFeature_RNA)) + 
  geom_histogram(binwidth = 100) + 
  ggtitle("nFeature_RNA") +
  theme_minimal() +
  #geom_hline(yintercept = 0) +
  xlab("genes per nucleus") +
  ylab("frequency") +
  theme(axis.title = element_text(size = 16),
        axis.text = element_text(size = 14))

p1 <- ggplot(NS_new_metadata, aes(x = nCount_RNA)) + 
  geom_histogram(binwidth = 2500) + 
  #coord_cartesian(xlim = c(0, 400000)) + 
  ggtitle("nCount_RNA") +
  theme_minimal() +
  #geom_hline(yintercept = 0) +
  xlab("UMIs per nucleus") +
  ylab("frequency") +
  theme(axis.title = element_text(size = 16),
        axis.text = element_text(size = 14))

plot_grid(p1,p2, nrow = 1, align = "hv", axis = "lrbt")
```


Figure S6b: UMIs and genes per nucleus across clusters
``` {r}
p3 <- nice_violin_plot(NS_new, features = "nCount_RNA", group_by = "cell_type", sort = F)
p4 <- nice_violin_plot(NS_new, features = "nFeature_RNA", group_by = "cell_type", sort = F) + coord_cartesian(ylim = c(0, 15000))
plot_grid(p3,p4, nrow = 1, align = "hv", axis = "lrbt")
```


Figure S6c: compare mean gene expression across scRNA-seq and snRNA-seq datasets
Determine mean expression of common genes:
``` {r}
common_genes <- intersect(rownames(NS_new), rownames(EF.filt))

expr_NS <- GetAssayData(NS_new, slot = "data", assay = "SCT") %>% 
  .[rownames(.) %in% common_genes,] %>% 
  data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  mutate(mean_NS = rowMeans(.[,2:ncol(.)])) %>% 
  dplyr::select(gene, mean_NS)

expr_EF <- GetAssayData(EF.filt, slot = "data", assay = "SCT") %>% 
  .[rownames(.) %in% common_genes,] %>% 
  data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  mutate(mean_SC = rowMeans(.[,2:ncol(.)])) %>% 
  dplyr::select(gene, mean_SC)

expr_both <- merge(x = expr_EF,
                   y = expr_NS,
                   by = "gene")
```

Linear regression:
``` {r}
lm_res <- lm(mean_SC ~ mean_NS, expr_both)
summary.lm(lm_res)
rsq <- summary(lm_res)$r.squared %>% round(digits = 4)
```

Find outliers to be labelled on plot:
``` {r}
expr_both$residuals <- residuals(lm_res)
ggplot(expr_both, aes(residuals)) + geom_histogram() # look at histogram to set residual threshold
```

Set outlier cutoff of 0.5 and -0.5:
``` {r}
outliers <- expr_both[expr_both$residuals >= 0.5 | expr_both$residuals <= -0.5, ]$gene
```

Scatter plot:
``` {r}
require(ggrepel)
ggplot(expr_both, aes(x = mean_SC, y = mean_NS)) +
  geom_point(size = 0.75) +
  #geom_smooth(method = "lm", color = "#B8122690", size = 2) +
  geom_text_repel(data = (expr_both[expr_both$gene %in% outliers,]),
                  aes(x = mean_SC, y = mean_NS, label = gene, size = 14),
                  segment.alpha = 0.5,
                  show.legend = F,
                  point.padding = 0.25,
                  min.segment.length = 0.05) +
  xlab("mean expression in scRNA-seq") +
  ylab("mean expression in snRNA-seq") +
  theme_minimal() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 16)) +
  coord_cartesian(xlim = c(0,6), ylim = c(0,6)) 
```


Figure S6d: compare mean gene expression of cilia genes across scRNA-seq and snRNA-seq datasets
Determine mean expression of cilia genes:
``` {r}
load("/Users/hunterm/Dropbox/MH_ST/Miranda_R/cilia/SYSCILIA_all_fish.R") # Load list of SYSCILIA fish genes

expr_NS <- GetAssayData(NS_new, slot = "data", assay = "SCT") %>% 
  .[rownames(.) %in% cilia_fish,] %>% 
  data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  mutate(mean_NS = rowMeans(.[,2:ncol(.)])) %>% 
  dplyr::select(gene, mean_NS)

expr_EF <- GetAssayData(EF.filt, slot = "data", assay = "SCT") %>% 
  .[rownames(.) %in% cilia_fish,] %>% 
  data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  mutate(mean_SC = rowMeans(.[,2:ncol(.)])) %>% 
  dplyr::select(gene, mean_SC)

expr_both <- merge(x = expr_EF,
                   y = expr_NS,
                   by = "gene")
```

Linear regression:
``` {r}
lm_res <- lm(mean_SC ~ mean_NS, expr_both)
summary.lm(lm_res)
rsq <- summary(lm_res)$r.squared %>% round(digits = 4)
```

Find outliers to be labelled on plot:
``` {r}
expr_both$residuals <- residuals(lm_res)
ggplot(expr_both, aes(residuals)) + geom_histogram() # look at histogram to set residual threshold
```

Set outlier cutoff of 0.5 and -0.5:
``` {r}
outliers <- expr_both[expr_both$residuals >= 0.5 | expr_both$residuals <= -0.5, ]$gene
```

Scatter plot:
``` {r}
require(ggrepel)
p2 <- ggplot(expr_both, aes(x = mean_SC, y = mean_NS)) +
  geom_point(size = 0.75) +
  #geom_smooth(method = "lm", color = "#B8122690", size = 2) +
  geom_text_repel(data = (expr_both[expr_both$gene %in% outliers,]),
                  aes(x = mean_SC, y = mean_NS, label = gene, size = 14),
                  segment.alpha = 0.5,
                  show.legend = F,
                  point.padding = 0.25,
                  min.segment.length = 0.05) +
  xlab("mean expression in scRNA-seq") +
  ylab("mean expression in snRNA-seq") +
  theme_minimal() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 16)) +
  coord_cartesian(ylim = c(0,3.5), xlim = c(0,3.5))
```

``` {r}
sessionInfo()
```








